ObjCVideoQuickstart/ViewController.m

```diff
@@ -0,0 +1,314 @@
+        
+        // We will set the prepared local media in connect options.
+        builder.localMedia = self.localMedia;
+        
+        // The name of the Room where the Client will attempt to connect to. Please note that if you pass an empty
+        // Room `name`, the Client will create one for you. You can get the name or sid from any connected Room.
+        builder.name = self.roomTextField.text;
+    }];
+    
+    // Attempting to connect to room with connect options.
+    self.room = [self.client connectWithOptions:connectOptions delegate:self];
+    
+    [self logMessage:[NSString stringWithFormat:@"Attempting to connect to room %@", self.roomTextField.text]];
+    
+    [self toggleView];
+    [self dismissKeyBoard];
+}
+
+- (IBAction)disconnectButtonPressed:(id)sender {
+    [self.room disconnect];
+}
+
+- (IBAction)micButtonPressed:(id)sender {
+    // We will toggle the mic to mute/unmute and change the title according to the user action. 
+    
+    if ([self.localMedia.audioTracks count] > 0) {
+        self.localMedia.audioTracks[0].enabled = !self.localMedia.audioTracks[0].isEnabled;
+        
+        //toggle the button title
+        if (self.localMedia.audioTracks[0].isEnabled) {
+            [self.micButton setTitle:@"Mute" forState:UIControlStateNormal];
+        } else {
+            [self.micButton setTitle:@"Unmute" forState:UIControlStateNormal];
+
+        }
+    }
+}
+
+// Reset the client ui status
+- (void)toggleView {
+    [self.micButton setTitle:@"Mute" forState:UIControlStateNormal];
+    
+    self.roomTextField.hidden = !self.roomTextField.isHidden;
+    self.connectButton.hidden = !self.connectButton.isHidden;
+    self.disconnectButton.hidden = !self.disconnectButton.isHidden;
+    self.roomLine.hidden = !self.roomLine.isHidden;
+    self.roomLabel.hidden = !self.roomLabel.isHidden;
+    self.micButton.hidden = !self.micButton.isHidden;
+    [UIApplication sharedApplication].idleTimerDisabled = ![UIApplication sharedApplication].isIdleTimerDisabled;
+}
+
+- (void)dismissKeyBoard {
+    if (self.roomTextField.isFirstResponder) {
+        [self.roomTextField resignFirstResponder];
+    }
+}
+
+- (BOOL)testFieldShouldReturn:(UITextField *)textField {
+    [self connectButtonPressed:textField];
+    return YES;
+}
+
+- (void)cleanupRemoteParticipant {
+    if (self.participant) {
+        if ([self.participant.media.videoTracks count] > 0) {
+            [self.participant.media.videoTracks[0] detach:self.remoteView];
+        }
+        self.participant = nil;
+    }
+}
+
+- (void)logMessage:(NSString *)msg {
+    self.messageLabel.text = msg;
+}
+
+#pragma mark - TVIRoomDelegate
+
+- (void)didConnectToRoom:(TVIRoom *)room {
+    // At the moment, this example only supports rendering one Participant at a time.
+    
+    [self logMessage:[NSString stringWithFormat:@"Connected to room %@", room.name]];
```

Comment:
You might want to take this opportunity to log the `LocalParticipant` name, and maybe even the number of participants already connected.